opentelemetry/proto/profiles/v1experimental/profiles.proto

```diff
@@ -120,42 +129,143 @@ message ResourceProfiles {
+  // sample_type.unit == "count".
+  repeated ValueType sample_type = 1;
+  // The set of samples recorded in this profile.
+  repeated Sample sample = 2;
+  // Mapping from address ranges to the image/binary/library mapped
+  // into that address range.  mapping_table[0] will be the main binary.
+  repeated Mapping mapping_table = 3;
+  // Locations referenced by samples via location_indices.
+  repeated Location location_table = 4;
+  // Functions referenced by locations.
+  repeated Function function_table = 5;
+  // A common table for strings referenced by various messages.
+  // string_table[0] must always be "".
+  repeated string string_table = 6;
+  // frames with Function.function_name fully matching the following
+  // regexp will be dropped from the samples, along with their successors.
+  int32 drop_frames_strindex = 7;   // Index into string table.
+  // frames with Function.function_name fully matching the following
+  // regexp will be kept, even if it matches drop_frames.
+  int32 keep_frames_strindex = 8;  // Index into string table.
+
+  // The following pprof fields are informational, do not affect
+  // interpretation of results.
+
+  // Time of collection (UTC) represented as nanoseconds past the epoch.
+  int64 time_nanos = 9;
+  // Duration of the profile, if a duration makes sense.
+  int64 duration_nanos = 10;
+  // The kind of events between sampled occurrences.
+  // e.g [ "cpu","cycles" ] or [ "heap","bytes" ]
+  ValueType period_type = 11;
+  // The number of events between sampled occurrences.
+  int64 period = 12;
+  // Free-form text associated with the profile. The text is displayed as is
+  // to the user by the tools that read profiles (e.g. by pprof). This field
+  // should not be used to store any machine-readable information, it is only
+  // for human-friendly content. The profile must stay functional if this field
+  // is cleaned.
+  repeated int32 comment_strindices = 13; // Indices into string table.
+  // Index into the string table of the type of the preferred sample
+  // value. If unset, clients should default to the last sample value.
+  int32 default_sample_type_strindex = 14;
+
+  reserved 15; // doc_url field is not inherited from pprof.
+
+  // In the event that future pprof versions add additional fields, then
+  // future OpenTelementry versions MAY use these for corresponding purposes.
+  reserved 16,17,18,19;
+
+
+  // second section: OpenTelemetry fields.
+
   // A globally unique identifier for a profile. The ID is a 16-byte array. An ID with
   // all zeroes is considered invalid.
   //
   // This field is required.
-  bytes profile_id = 1;
+  bytes profile_id = 20;
 
   // start_time_unix_nano is the start time of the profile.
   // Value is UNIX Epoch time in nanoseconds since 00:00:00 UTC on 1 January 1970.
   //
   // This field is semantically required and it is expected that end_time >= start_time.
-  fixed64 start_time_unix_nano = 2;
+  fixed64 start_time_unix_nano = 21;
```

Comment:
There are already time_nanos and duration_nanos fields above. Can we just use those?

FWIW I have a slight preference to start+duration rather than start+end:
* During debugging it's useful to see the duration easily: "is this a 10 second profile or maybe zero duration profile like heap?"
* Duration is a smaller value and is encoded more effectively. This is a more minor thing since this value is just stored once.

Reply:
Makes sense. Done.